sync/metrics.go

```diff
@@ -24,27 +37,103 @@ func newMetrics() (*metrics, error) {
 		return nil, err
 	}
 
+	syncLoopStarted, err := meter.Int64Counter(
+		"sync_loop_started",
+		metric.WithDescription("sync loop started"),
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	trustedPeersOutOfSync, err := meter.Int64Counter(
+		"tr_peers_out_of_sync",
+		metric.WithDescription("trusted peers out of sync"),
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	laggingHeadersStart, err := meter.Int64Counter(
+		"sync_lagging_hdr_start",
+		metric.WithDescription("lagging header start"),
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	subjectiveHead, err := meter.Int64ObservableGauge(
+		"sync_subjective_head",
+		metric.WithDescription("subjective head height"),
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	blockTime, err := meter.Float64Histogram(
+		"sync_actual_blockTime_ts",
+		metric.WithDescription("duration between creation of 2 blocks"),
+	)
+	if err != nil {
+		return nil, err
+	}
+
 	m := &metrics{
-		totalSyncedGauge: totalSynced,
+		totalSyncedGauge:      totalSynced,
+		syncLoopStarted:       syncLoopStarted,
+		trustedPeersOutOfSync: trustedPeersOutOfSync,
+		laggingHeadersStart:   laggingHeadersStart,
+		blockTime:             blockTime,
+		headersThreshold:      headersThreshold,
 	}
 
 	callback := func(ctx context.Context, observer metric.Observer) error {
 		observer.ObserveFloat64(totalSynced, float64(m.totalSynced.Load()))
+		observer.ObserveInt64(subjectiveHead, m.subjectiveHead.Load())
 		return nil
 	}
-	_, err = meter.RegisterCallback(callback, totalSynced)
+
+	_, err = meter.RegisterCallback(callback, totalSynced, subjectiveHead)
 	if err != nil {
 		return nil, err
 	}
-
 	return m, nil
 }
 
-// recordTotalSynced records the total amount of synced headers.
 func (m *metrics) recordTotalSynced(totalSynced int) {
 	if m == nil {
 		return
 	}
-
 	m.totalSynced.Add(int64(totalSynced))
 }
+
+func (m *metrics) recordSyncLoopStarted() {
```

Comment:
Naming nit: we can remove the `record` word from all the methods. It's redundant and does not bring readability as it's always called on the `metrics` field. This will also make it consistent with other meters in go-header